rkt/image/fetchers/https/main.go

```diff
@@ -0,0 +1,213 @@
+	"strings"
+	"time"
+
+	"github.com/coreos/pkg/progressutil"
+	"github.com/coreos/rkt/rkt/image/fetchers"
+)
+
+// statusAcceptedError is an error returned when resumableSession
+// receives a 202 HTTP status. It is mostly used for deferring
+// signature downloads.
+type statusAcceptedError struct{}
+
+func (statusAcceptedError) Error() string { return "download deferred" }
+
+func errAndExit(format string, a ...interface{}) {
+	stderr(format, a...)
+	os.Exit(1)
+}
+
+func stderr(format string, a ...interface{}) {
+	out := fmt.Sprintf(format, a...)
+	fmt.Fprintln(os.Stderr, strings.TrimSuffix(out, "\n"))
+}
+
+func main() {
+	config, err := fetchers.ConfigFromStdin()
+	if err != nil {
+		errAndExit("error reading config: %v", err)
+	}
+
+	if config.Version != 1 {
+		errAndExit("unsupported plugin schema version")
+	}
+
+	err = fetch(config)
+	if err != nil {
+		errAndExit("error fetching http(s) image: %v", err)
+	}
+}
+
+func fetch(config *fetchers.Config) error {
+	if config.InsecureOpts.SkipTLSCheck {
+		stderr("warning: TLS verification has been disabled")
+	}
+	if config.InsecureOpts.SkipImageCheck {
+		stderr("warning: image signature verification has been disabled")
+	}
+	if config.InsecureOpts.AllowHTTP {
+		stderr("warning: image allowed to be fetched without encryption")
+	}
+	if !config.InsecureOpts.AllowHTTP && config.Scheme == "http" {
+		return fmt.Errorf("error: http URLs not allowed")
+	}
+	retry := false
+	if !config.InsecureOpts.SkipImageCheck {
+		var err error
+		retry, err = fetchSignature(config)
+		if err != nil {
+			return err
+		}
+	}
+	res, err := fetchImage(config)
+	if err != nil {
+		return err
+	}
+	if !retry {
+		return fetchers.ResultToStdout(res)
+	}
+	retry, err = fetchSignature(config)
+	if err != nil {
+		return err
+	}
+	if retry {
+		return fmt.Errorf("error downloading the signature file: server asked to defer the download again")
+	}
+	return fetchers.ResultToStdout(res)
+}
+
+func fetchImage(config *fetchers.Config) (*fetchers.Result, error) {
+	aciFile, err := os.Create(config.OutputACIPath)
+	if err != nil {
+		return nil, err
+	}
+	defer aciFile.Close()
+	return fetchURL(config, "ACI", config.Scheme+"://"+config.Name, aciFile)
+}
+
+func fetchSignature(config *fetchers.Config) (bool, error) {
+	ascFile, err := os.Create(config.OutputASCPath)
+	if err != nil {
+		return false, err
+	}
+	defer ascFile.Close()
+	_, err = fetchURL(config, "Signature", config.Scheme+"://"+config.Name+".asc", ascFile)
+	if err == nil {
+		return false, nil
+	}
+	if _, ok := err.(*statusAcceptedError); ok {
+		stderr("server requested deferring the signature download")
+		return true, nil
+	}
+	return false, fmt.Errorf("error downloading the signature file: %v", err)
+}
+
+func fetchURL(config *fetchers.Config, name, u string, dest io.Writer) (*fetchers.Result, error) {
+	stderr("fetching from URL: %s", u)
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: config.InsecureOpts.SkipTLSCheck},
+	}
```

Comment:
Isn't dropping the original httpfetcher implementation with this one losing the ability to resume a download (and probably other features)?